### PR TITLE
feat(block-section): add `icon` property (deprecates `status`)

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #8110

## Summary
Remove the `status` property and instead add an `icon` property.

Provide a demo and screenshot coverage.

### CODE REVIEW

File: packages/calcite-components/src/components/block-section/block-section.scss
Comment (jcfranco) [inline packages/calcite-components/src/components/block-section/block-section.scss:1]: We might want to hold off on documenting these. The review from https://github.com/Esri/calcite-design-system/pull/8594 might affect some props. If we want to be extra careful, we might need to rename these as internal.

Comment (Elijbet) [reply packages/calcite-components/src/components/block-section/block-section.scss:1]: Sure, that makes sense.

File: packages/calcite-components/src/components/block-section/block-section.tsx
Comment (jcfranco) [inline packages/calcite-components/src/components/block-section/block-section.tsx:218]: To simplify a bit, can you look at adding a `renderIcon` method that returns an icon with the corresponding start/end props?

File: packages/calcite-components/src/components/block-section/block-section.tsx
Comment (jcfranco) [inline packages/calcite-components/src/components/block-section/block-section.tsx:223]: Does this need to use the [`getIconScale`](https://github.com/Esri/calcite-design-system/blob/f1ff5e1acd0e802f6c5b635a77dedf8138d7e753/packages/calcite-components/src/utils/component.ts#L4-L6) util?

File: packages/calcite-components/src/components/block-section/block-section.tsx
Comment (jcfranco) [inline packages/calcite-components/src/components/block-section/block-section.tsx:64]: Can you make sure props are in alphabetical order?

**Review by jcfranco — APPROVED** [review]
Looks great! A few things:

* Let's target this for the May release. The [token work](#8594) might affect it and it'll also give more time for users to prep for the visual changes this will introduce. 
* Can you apply the `visual changes` to the issue?
* Can you rephrase the PR title to be more along the lines of: `feat(component): add X prop (deprecates Y)`? We want to highlight the added prop first to align with the commit type.

**Review by jcfranco — APPROVED** [review]
Looks great! A few things:

* Let's target this for the May release. The [token work](#8594) might affect it and it'll also give more time for users to prep for the visual changes this will introduce. 
* Can you apply the `visual changes` to the issue?
* Can you rephrase the PR title to be more along the lines of: `feat(component): add X prop (deprecates Y)`? We want to highlight the added prop first to align with the commit type.

File: packages/calcite-components/src/components/block-section/block-section.tsx
Comment (Elijbet) [reply packages/calcite-components/src/components/block-section/block-section.tsx:223]: `block/block-section` don't have `scale` currently. It'll be reviewed in https://github.com/Esri/calcite-design-system/issues/7082. As is, we think of the components that don't have scale as `medium` and use `small` icons for them.

File: packages/calcite-components/src/components/block-section/block-section.stories.ts
Comment (jcfranco) [inline packages/calcite-components/src/components/block-section/block-section.stories.ts:8]: Can you move this into the block stories? We usually include child stories to the parent one.

File: packages/calcite-components/src/components/block-section/block-section.tsx
Comment (jcfranco) [inline packages/calcite-components/src/components/block-section/block-section.tsx:268]: Let's add a `label` prop to the CSS lookup object.

File: packages/calcite-components/src/components/block-section/block-section.scss
Comment (jcfranco) [inline packages/calcite-components/src/components/block-section/block-section.scss:75]: Let's keep this if the internal CSS props help with #8594 integration. Otherwise, it might be cleaner to remove these here and add when the PR is synced w/ `main`.

**Review by jcfranco — APPROVED** [review]
Good to merge once last round of changes are addressed. 🎉

**Pull request merged**
